packages/backend/src/peripherals/database/WithdrawableAssetRepository.ts

```diff
@@ -63,6 +66,19 @@ export class WithdrawableAssetRepository extends BaseRepository {
     return results[0]!.id
   }
 
+  async getAssetBalancesByStarkKey(
+    starkKey: StarkKey
+  ): Promise<{ assetHash: AssetHash; balanceDelta: bigint }[]> {
+    const knex = await this.knex()
+    const results: { assetHash: AssetHash; balanceDelta: bigint }[] =
+      await knex('withdrawable_assets')
+        .select('asset_hash as assetHash')
+        .where('stark_key', starkKey.toString())
+        .groupBy('asset_hash')
+        .sum('balance_delta as balanceDelta')
+    return results.filter((result) => result.balanceDelta > 0)
```

Comment:
Why is filtering with SQL better?

Reply:
Because otherwise the database returns values for all users from the beginning of time, even though sums of 99.999% of them are 0. So you get big response from DB that you later filter in JS. It would be more efficient to just get the records you're interested in.